x/feemarket/post/fee.go

```diff
@@ -212,10 +215,6 @@ func (dfd FeeMarketDeductDecorator) DeductFeeAndTip(ctx sdk.Context, sdkTx sdk.T
 // DeductCoins deducts coins from the given account.
 // Coins can be sent to the default fee collector (causes coins to be distributed to stakers) or sent to the feemarket fee collector account (causes coins to be burned).
 func DeductCoins(bankKeeper BankKeeper, ctx sdk.Context, acc sdk.AccountI, coins sdk.Coins, distributeFees bool) error {
-	if !coins.IsValid() {
```

Comment:
Can you please elaborate this decision, it's not super clear to me.

Reply:
The coins are already guaranteed to be valid from how we are constructing them.  The only additional check this was doing was ensuring they are `non-zero`